From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pragyaan Gaur <pragyaangaur12@gmail.com>
Date: Mon, 12 Oct 2026 13:06:52 +0530
Subject: [PATCH 2/2] fix(airconv): pass the memory flags argument to the
 atomic add

Newer Metal compilers declare __metal_atomic_fetch_add_explicit with a fifth memory flags argument, and the shader library fails to build without it.
---
 src/airconv/shaders/air_tessellation.metal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/airconv/shaders/air_tessellation.metal b/src/airconv/shaders/air_tessellation.metal
index 56ba86e..078304f 100644
--- a/src/airconv/shaders/air_tessellation.metal
+++ b/src/airconv/shaders/air_tessellation.metal
@@ -150,7 +150,7 @@ struct TessMeshWorkload {
 
 int
 get_next_index(threadgroup int *out_count) {
-  return __metal_atomic_fetch_add_explicit(out_count, 1, int(memory_order_relaxed), __METAL_MEMORY_SCOPE_THREADGROUP__);
+  return __metal_atomic_fetch_add_explicit(out_count, 1, int(memory_order_relaxed), __METAL_MEMORY_SCOPE_THREADGROUP__, __METAL_MEMORY_FLAGS_NONE__);
 }
 
 template <partitioning partition>
